roles/container-engine/cri-o/tasks/main.yaml

```
@@ -55,6 +55,45 @@
   when:
     - youki_enabled
 
+- name: Cri-o | read default container runtime from crio.conf
+  ansible.builtin.shell: |
+    set -o pipefail && grep "^default_runtime" /etc/crio/crio.conf | awk -F'=' '{print $2}' | xargs
+  register: existing_crio_default_runtime
+  when: ansible_facts.services['crio.service'] is defined
+
+- name: Cri-o | set existing crio default runtime fact
+  ansible.builtin.set_fact:
+    existing_crio_default_runtime_output: "{{ existing_crio_default_runtime.stdout }}"
+  when: ansible_facts.services['crio.service'] is defined
+
+- name: Cri-o | compare existing default runtime with default_runtime
+  ansible.builtin.set_fact:
+    runtime_matches: "{{ existing_crio_default_runtime_output is equalto(crio_default_runtime) }}"
+  when: ansible_facts.services['crio.service'] is defined
+
+- name: Cri-o | Stop kubelet service if running
+  service:
+    name: kubelet
+    state: stopped
+  when:
+    - runtime_matches is defined and runtime_matches is false
+    - ansible_facts.services['kubelet.service'] is defined and ansible_facts.services['kubelet.service'].state == 'running'
+
+- name: Cri-o | Stop all the running containers
+  ansible.builtin.shell: |
+    set -o pipefail && {{ bin_dir }}/crictl ps -q | xargs -r {{ bin_dir }}/crictl stop
+  when:
+    - runtime_matches is defined and runtime_matches is false
+    - ansible_facts.services['crio.service'] is defined
```

Technically you'd only want to stop the containers using the default runtime (which probably translate to `runtimeHandler: ""` in the output of `crictcl ps -o json`.

Filtering on that is not necessary for this PR, but doing in a follow-up would be appreciated...

But same as above, please avoid shell here for making it easier afterwards
```suggestion
- name: Cri-o | List running containers
  ansible.builtin.command: {{ bin_dir }}/crictl ps -o json
  changed_when: false
  register: running_containers
  
  - name: Cri-o | List running containers
  ansible.builtin.command: {{ bin_dir }}/crictl stop  {{ (running_containers.stoud | from_json).containers | map(attr='id') # something like that
```